server/src/main/java/org/opensearch/threadpool/ThreadPool.java

```diff
@@ -696,7 +695,13 @@ public Info(String name, ThreadPoolType type, int min, int max, @Nullable TimeVa
 
         public Info(StreamInput in) throws IOException {
             name = in.readString();
-            type = ThreadPoolType.fromType(in.readString());
+            final String typeStr = in.readString();
+            // Opensearch on or after 3.0.0 version doesn't know about "fixed_auto_queue_size" thread pool. Convert it to RESIZABLE.
+            if (typeStr.equalsIgnoreCase("fixed_auto_queue_size")) {
+                type = ThreadPoolType.RESIZABLE;
```

Comment:
Just curious, but why resizable instead of fixed?

Reply:
Thanks @andrross, technically `RESIZABLE` is better fit, that is the one we introduced instead here [1], but `fixed` would also work, thank you! 

[1] https://github.com/opensearch-project/OpenSearch/pull/3207

Reply:
Ah, I think I misunderstood. "Resizable" refers to the queue being resizable, not the size of the thread pool itself. That's definitely a better fit to replace "fixed_auto_queue_size".

Reply:
Exactly, thank you!